alica_viewer/include/alica_viewer/alica_plan_parser.h

```diff
@@ -0,0 +1,277 @@
+#ifndef ALICA_VIEWER_ALICA_PLAN_PARSER_H
+#define ALICA_VIEWER_ALICA_PLAN_PARSER_H
+
+#include <SystemConfig.h>
+#include <engine/PlanRepository.h>
+#include <engine/SimplePlanTree.h>
+#include <engine/Types.h>
+#include <engine/containers/PlanTreeInfo.h>
+#include <engine/model/EntryPoint.h>
+#include <engine/model/Plan.h>
+#include <engine/model/State.h>
+#include <engine/model/Task.h>
+#include <engine/parser/PlanParser.h>
+#include <supplementary/AgentID.h>
+#include <supplementary/AgentIDManager.h>
+
+namespace alica
+{
+
+typedef std::unordered_map<const supplementary::AgentID*, std::shared_ptr<SimplePlanTree>, supplementary::AgentIDHash, supplementary::AgentIDEqualsComparator>
+    SimplePlanTreeMap;
+
+class AgentInfo
+{
+  public:
+    AgentInfo() {}
+
+    AgentInfo(int tmp_id, const std::string& tmp_name)
+        : id(tmp_id)
+        , name(tmp_name)
+    {
+    }
+
+    int id;
+    std::string name;
+};
+typedef std::unordered_map<const supplementary::AgentID*, AgentInfo, supplementary::AgentIDHash, supplementary::AgentIDEqualsComparator> AgentInfoMap;
+
+class CombinedSimplePlanTree
+{
+  public:
+    CombinedSimplePlanTree(const std::shared_ptr<SimplePlanTree> spt)
+        : _spt(spt)
+    {
+        addRobot(spt->getRobotId());
+    }
+
+    const std::unordered_set<std::shared_ptr<CombinedSimplePlanTree>>& getChildren() const { return _children; }
+
+    std::unordered_set<std::shared_ptr<CombinedSimplePlanTree>>& editChildren() { return _children; }
+
+    const std::shared_ptr<SimplePlanTree> getSpt() const { return _spt; }
+
+    const AgentGrp& getRobotsSorted()
+    {
+        std::sort(_robotIds.begin(), _robotIds.end(), supplementary::AgentIDComparator());
+        return _robotIds;
+    }
+
+    void addRobot(const supplementary::AgentID* robotId)
+    {
+        if (robotId) {
+            _robotIds.push_back(robotId);
+        }
+    }
+
+  private:
+    const std::shared_ptr<SimplePlanTree> _spt;
+    std::unordered_set<std::shared_ptr<CombinedSimplePlanTree>> _children;
```

Comment:
Why not just a vector of bare pointers? ownership should reside with the parent, and a vector is much faster for small sizes like this.